virttest/qemu_devices/utils.py

```diff
@@ -73,3 +73,16 @@ def none_or_int(value):
         return int(value)
     else:
         raise TypeError("This parameter has to be int or none")
+
+
+def set_cmdline_format_by_cfg(dev, config, key):
```

Comment:
Do we need to expose this function as a public API? I mean it makes sense for the level of the caller from the test case?
From my understanding, I think it just is useful for creating devices during calling `vm.create`, if I misunderstand, please point out it, thanks.
I suggest that we can assign it as a private API in the module `qcontainer`.

Reply:
> Do we need to expose this function as a public API? I mean it makes sense for the level of the caller from the test case? From my understanding, I think it just is useful for creating devices during calling `vm.create`, if I misunderstand, please point out it, thanks. I suggest that we can assign it as a private API in the module `qcontainer`.

From my opinion, the function may belong to the module `utils` ( This function has been used at `qemu_vm.py` and `qcontainer.py` ). Obviously, it's a shared function.  
In addition, once this function was declared as private, how other developers call it at other .py file?

Reply:
> > Do we need to expose this function as a public API? I mean it makes sense for the level of the caller from the test case? From my understanding, I think it just is useful for creating devices during calling `vm.create`, if I misunderstand, please point out it, thanks. I suggest that we can assign it as a private API in the module `qcontainer`.
> 
> From my opinion, the function may belong to the module `utils` ( This function has been used at `qemu_vm.py` and `qcontainer.py` ). Obviously, it's a shared function. In addition, once this function was declared as private, how other developers call it at other .py file?

Yes, I am afraid that other .py file call it in an invalid way since this function will change the `dev.cmdline_format`, I think it can not be exposed to the case level at least.

Reply:
> > > Do we need to expose this function as a public API? I mean it makes sense for the level of the caller from the test case? From my understanding, I think it just is useful for creating devices during calling `vm.create`, if I misunderstand, please point out it, thanks. I suggest that we can assign it as a private API in the module `qcontainer`.
> > 
> > 
> > From my opinion, the function may belong to the module `utils` ( This function has been used at `qemu_vm.py` and `qcontainer.py` ). Obviously, it's a shared function. In addition, once this function was declared as private, how other developers call it at other .py file?
> 
> Yes, I am afraid that other .py file call it in an invalid way since this function will change the `dev.cmdline_format`, I think it can not be exposed to the case level at least.

Basically, the 'dev' is an argument passed obviously by user. And if user uses this function, the user has to get the 'dev' first. 
In case level, if user really use this function, I believe that's a necessary test step.

Reply:
> > > > Do we need to expose this function as a public API? I mean it makes sense for the level of the caller from the test case? From my understanding, I think it just is useful for creating devices during calling `vm.create`, if I misunderstand, please point out it, thanks. I suggest that we can assign it as a private API in the module `qcontainer`.
> > > 
> > > 
> > > From my opinion, the function may belong to the module `utils` ( This function has been used at `qemu_vm.py` and `qcontainer.py` ). Obviously, it's a shared function. In addition, once this function was declared as private, how other developers call it at other .py file?
> > 
> > 
> > Yes, I am afraid that other .py file call it in an invalid way since this function will change the `dev.cmdline_format`, I think it can not be exposed to the case level at least.
> 
> Basically, the 'dev' is an argument passed obviously by user. And if user uses this function, the user has to get the 'dev' first. In case level, if user really use this function, I believe that's a necessary test step.

Yes, I agree with you but what kind of scenario needs the user call this function?

Reply:
> > > > > Do we need to expose this function as a public API? I mean it makes sense for the level of the caller from the test case? From my understanding, I think it just is useful for creating devices during calling `vm.create`, if I misunderstand, please point out it, thanks. I suggest that we can assign it as a private API in the module `qcontainer`.
> > > > 
> > > > 
> > > > From my opinion, the function may belong to the module `utils` ( This function has been used at `qemu_vm.py` and `qcontainer.py` ). Obviously, it's a shared function. In addition, once this function was declared as private, how other developers call it at other .py file?
> > > 
> > > 
> > > Yes, I am afraid that other .py file call it in an invalid way since this function will change the `dev.cmdline_format`, I think it can not be exposed to the case level at least.
> > 
> > 
> > Basically, the 'dev' is an argument passed obviously by user. And if user uses this function, the user has to get the 'dev' first. In case level, if user really use this function, I believe that's a necessary test step.
> 
> Yes, I agree with you but what kind of scenario needs the user call this function?

It depends on owner, I'm NOT sure which scenario calls this function.
On top of that, this function can't be declared to private in the 'qcontainer.py' since both the 'qemu_vm.py' and the 'qcontainer.py' call this function.